### DISCUSSION TITLE
Performance related to caching in Directus

### DISCUSSION THREAD

**Discussion by karmingc** [directus/directus#16242] [Archive]
Hello,

We currently use Redis as an in-memory cache store and we recently upgraded to a larger ec2 instances with more RAM while keeping the same configuration.
Noticeably, the CPU and Memory usage dropped significantly  in the instances hosting the directus application and I am trying to understand why. Without too much context on the intricacies of the caching layer, would there be any hypothesis to this?

**Comment by rijkvanzanten** [comment]
> we recently upgraded to a larger ec2 instances with more RAM while keeping the same configuration

If you increase the amount of available CPU and RAM on your machine while keeping everything else configured the same, it makes sense to see a drop in overall percentage of CPU and Memory used right? 🤔

**Reply by karmingc** [reply]
Sorry, I think I forgot to indicate that we saw the drop in the instances hosting the directus app, which is separate from the Redis instance.

**Reply by karmingc** [reply]
Likewise, we did notice that there was a drop in the # of requests as well, which could be explained by less retries.

I feel that that there might be something related to the connections...The only configuration I saw was the `CACHE_HEALTHCHECK_THRESHOLD`, which timed out on a few occasions. Other than that, I wasn't able to find out much concerning the connection pool, connection timeout and possibly other connection configurations on the Directus side for Redis.

**Reply by rijkvanzanten** [reply]
Oh wait am I misunderstanding? Are you saying you updated the Redis cluster, but left everything in Directus as-is?

**Reply by karmingc** [reply]
Yes, sorry for the confusion..

We noticed that the `/server/health` endpoint hits dropped by around 70%, which was mainly triggered by our k8s readiness probes. The larger Redis instance had no instance of page swapping, so I'm assuming that there's some possibilities of less retries.

What I'm a bit confused about is why the CPU/Memory on the Directus instance dropped significantly. We didn't see that many new connections and hypothetically Directus should be using a connection pool...

**Reply by rijkvanzanten** [reply]
Hard to tell exactly without some tracing and hands on debugging. It could very well be that the faster Redis instance is able to handle more data faster, which in turn means that node spends less time waiting for the data IO on that front, which in turn means less resources are used on the box on the connection itself

**Reply by karmingc** [reply]
Agreed. No worries, thanks @rijkvanzanten.
